src/PolarSSL++/BlockingSslClientSocket.cpp

```diff
@@ -182,7 +183,11 @@ bool cBlockingSslClientSocket::SetTrustedRootCertsFromString(const AString & a_C
 
 bool cBlockingSslClientSocket::Send(const void * a_Data, size_t a_NumBytes)
 {
-	ASSERT(m_IsConnected);
+	if (!m_IsConnected)
+	{
+		m_LastErrorText = "Socket is closed";
+		return false;
+	}
```

Comment:
Socket closure will be detected from within the `m_Ssl.WritePlain()`, it will return an error code, thus taking the branch on line 202. The error message will be different, but it will still be a failure.
This here is only so that the error message is obvious in the obvious cases.